code/parachain/frame/pablo/src/lib.rs

```diff
@@ -1080,52 +1081,60 @@ pub mod pallet {
 		) -> Result<SwapResult<Self::AssetId, Self::Balance>, DispatchError> {
 			let pool = Self::get_pool(pool_id)?;
 			let pool_account = Self::account_id(&pool_id);
-			let (base_amount, owner, fees) = match pool {
+			let (swap_result, fee, owner) = match pool {
 				PoolConfiguration::DualAssetConstantProduct(info) => {
-					// NOTE: lp_fees includes owner_fees.
-					let (base_amount, quote_amount_excluding_lp_fee, fees) =
-						DualAssetConstantProduct::<T>::do_swap(
-							&info,
-							&pool_account,
-							in_asset,
-							min_receive,
-							true,
-						)?;
+					let swap_result = Self::spot_price(pool_id, in_asset, min_receive.asset_id)?;
 
 					ensure!(
-						base_amount >= min_receive.amount,
+						swap_result.value.amount >= min_receive.amount,
 						Error::<T>::CannotRespectMinimumRequested
 					);
+					ensure!(
+						T::Assets::balance(min_receive.asset_id, &pool_account) >
+							swap_result.value.amount,
+						Error::<T>::NotEnoughLiquidity
+					);
 
+					// Transfer `in_asset_amount - fee_amount` to pool
 					T::Assets::transfer(
 						in_asset.asset_id,
 						who,
 						&pool_account,
-						quote_amount_excluding_lp_fee,
+						// NOTE: This needs better handling if we're to conditionally apply fees.
+						// One option is to return the correct amount to transfer to the pool from
+						// `spot_price`.
```

Comment:
The current interface does not support this. Should I go ahead and make the changes so that `spot_price` will return the pool's delta after fees in addition to what it already returns?

How do we plan to pass the value for applying fees or not? The current interface does not support it. Should changes be made elsewhere to enable this? @vimukthi-git